datafusion/physical-plan/src/projection.rs

```diff
@@ -357,6 +359,27 @@ impl ExecutionPlan for ProjectionExec {
     ) -> Result<FilterPushdownPropagation<Arc<dyn ExecutionPlan>>> {
         Ok(FilterPushdownPropagation::if_all(child_pushdown_result))
     }
+
+    fn try_pushdown_sort(
+        &self,
+        order: &[PhysicalSortExpr],
```

Comment:
What if the sort depends on the output of the projection? Presumably then the projection should say the sort can't be pushed down?